### PR TITLE
Support predefined sizes in legacy Image and Picture

### LINKED ISSUE #792: Legacy image and picture classes do not support predefined image sizes (via config)
**Description**  
`\Contao\Image::create()` does not support image sizes defined via configuration.

Only numeric values are accepted and resolved as entries of `tl_image_size`:
https://github.com/contao/contao/blob/f6997ffb142cc6c69eba6e647cde303deec68f60/core-bundle/src/Resources/contao/library/Contao/Image.php#L767-L772

Imo we should either add that functionality or instead use the `contao.image.image_factory` service in this function. Otherwise 3rd party extensions like [isotope](https://github.com/isotope/core/issues/2081) won't work.

(The same holds true for `\Contao\Picture`.)

### PR DESCRIPTION
**Pull request by ausi**
Fixes #792

@m-vo Please verify if this fix works for your use case.

### CODE REVIEW

File: core-bundle/src/Resources/contao/library/Contao/Image.php [hunk 1]
```diff
@@ -769,6 +769,12 @@ public static function create($image, $size=null)
 			$size = (int) $size[2];
 		}
 
+		// predefined image size as resize mode
+		elseif (\is_array($size) && !empty($size[2]) && is_string($size[2]) && $size[2][0] === '_')
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/library/Contao/Image.php]: There already is an `is_array($size)` check in line 767 and another one in line 778. Does this need to be refactored?

File: core-bundle/src/Resources/contao/library/Contao/Picture.php [hunk 2]
```diff
@@ -103,9 +103,15 @@ public static function create($file, $size=null)
 			$size = (int) $size[2];
 		}
 
+		// predefined image size as resize mode
+		elseif (\is_array($size) && !empty($size[2]) && is_string($size[2]) && $size[2][0] === '_')
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/library/Contao/Picture.php]: The same here.

**Review by leofeyer — CHANGES_REQUESTED** [review]

File: core-bundle/src/Resources/contao/library/Contao/Image.php (hunk 1 above)
Comment (ausi) [reply core-bundle/src/Resources/contao/library/Contao/Image.php]: Done in e97bba284687fa9c6db236f032eebf2238922d6a

File: core-bundle/src/Resources/contao/library/Contao/Picture.php (hunk 2 above)
Comment (ausi) [reply core-bundle/src/Resources/contao/library/Contao/Picture.php]: Done in e97bba284687fa9c6db236f032eebf2238922d6a

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @ausi.

**Comment by m-vo** [comment]
I'm a bit late to this party but still: The fix works in my use case. :tada:

Thanks @ausi!

**Pull request merged**